internal/resource/http.go

```diff
@@ -109,27 +109,45 @@ func (f *Fetcher) UpdateHttpTimeoutsAndCAs(timeouts types.Timeouts, cas []types.
 		if err != nil {
 			return err
 		}
-		block, _ := pem.Decode(cablob)
-		if block == nil {
-			f.Logger.Err("Unable to decode CA (%v)", ca.Source)
-			return ErrPEMDecodeFailed
-		}
-
-		cert, err := x509.ParseCertificate(block.Bytes)
+		// For Root CA
+		restCAs, err := f.parseCaBundle(cablob, ca, pool)
 		if err != nil {
-			f.Logger.Err("Unable to parse CA (%v): %s", ca.Source, err)
+			f.Logger.Err("Unable to parse CA bundle: %s", err)
 			return err
 		}
-
-		f.Logger.Info("Adding %q to list of CAs", cert.Subject.CommonName)
-		pool.AddCert(cert)
+		// For Intermediate CAs
+		for len(restCAs) > 0 {
+			res, err := f.parseCaBundle(restCAs, ca, pool)
+			if err != nil {
+				f.Logger.Err("Unable to parse CA bundle: %s", err)
+				return err
+			}
+			restCAs = res
```

Comment:
can someone that knows golang better than I confirm this is a safe way to extend this for loop?

Reply:
Should be.